### DISCUSSION TITLE
Why Axios throws exceptions for status code not equal to 200?

### DISCUSSION THREAD

**Discussion by stefanodecillis** [axios/axios#5818] [General]
As in the title of the discussion, why Axios throws exceptions in this way? With this approach, we need to catch each status not equal to 200. Usually, the mapping of status codes for responses should not be handled by the http client. The application should do it. Moreover, Exceptions are really important when developing the application and catching random exceptions makes no sense (always in my opinion). I see this behavior as an abuse of exceptions

**Comment by gusgorman2** [comment]
I am also very confused by this. The axios API makes it look at as if you can check the response status code, for example, it looks as if you can do this:

```
const response: AxiosResponse<T, any> = await axios.get(url)
if (response.status === HttpStatusCode.Ok) {
      return response.data
} else {
      // check other status codes here
}
```
However, in practice as the OP as noted, this never happens, and an exception gets thrown. This is very misleading. Or am I missing something?

**Reply by gusgorman2** [reply]
Looking at the docs, I now see this is configurable:

https://axios-http.com/docs/handling_errors

> Using the validateStatus config option, you can define HTTP code(s) that should throw an error.
> 
> axios.get('/user/12345', {
>   validateStatus: function (status) {
>     return status < 500; // Resolve only if the status code is less than 500
>   }
> })

However, I can't see immediately what the default behaviour is. From a bit of testing it seems to be returning for 200 and maybe 400, but throwing for everything else ?

**Comment by igorsantos07** [comment]
It's sad to see such an important library changing such an important behavior without proper documentation. This is very much a breaking change, and one I missed because there's no clear docs on the changes from 0.x to 1.x. Now I'm stuck with an old way to handle errors and no clear path as how to fix it, as my application centralizes common errors codes in a single interceptor, and now it seems I also need to `catch()` errors everywhere, even if I decide to handle them in an interceptor.

I opened an issue to request a docs page on the migration: https://github.com/axios/axios/issues/7208
